Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,139 @@
+                return
+            }
+            
+            for item in body.details.displayItems {
+                popup.addDisplayItemLabel(message: "\(item.label): \(item.amount.value) \(item.amount.currency)\n")
+                log.info(item.label)
+            }
+            
+            popup.addTotalLabel(message: "\(body.details.total.label):  \(body.details.total.amount.value) \(body.details.total.amount.currency)\n")
+            
+            popup.addButton(title: Strings.paymentRequestPay) { [weak self] in
+                guard let self = self else {
+                    return .flyDown
+                }
+                guard let rewards = self.tab?.rewards, let publisher = self.tab?.publisher, let amount = Double(body.details.total.amount.value) else {
+                    return .flyDown
+                }
+                
+                // To-Do: Replace tipping code with SKU APIs - https://github.com/brave/brave-ios/issues/2383
+                rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                    self.response = """
+                        {
+                          "requestId": "a62c29b3-f840-47cd-b895-4573d3190227",
+                          "methodName": "bat",
+                          "details": {
+                            "transaction_id": "bcbbd947-346d-439f-96b4-101bbd966675",
+                            "message": "Payment for Ethiopian Coffee!"
+                          }
+                        }
+                    """
+                    
+                    ensureMainThread {
+                        let trimmed = self.response.removingNewlines()
+                        self.tab?.webView?.evaluateJavaScript("PaymentRequestCallback\(self.token).paymentreq_postCreate('\(trimmed)', '', '')") { _, error in
+                                if error != nil {
+                                    log.error(error)
+                                }
+                            }
+                    }
+                }
+                
+                return .flyDown
+            }
+            
+            popup.addButton(title: Strings.paymentRequestCancel) { [weak self] in
+                guard let self = self else {
+                    return .flyDown
+                }
+                
+                ensureMainThread {
+                    self.sendPaymentRequestError(errorName: PaymentRequestErrors.AbortError.rawValue, errorMessage: Strings.userCancelledMessage)
+                }
+                
+                return .flyDown
+            }
+            popup.showWithType(showType: .flyUp)
+        } catch {
+            log.error(error)
+        }
+            
+    }
+}
+
+extension String {
+    func removingNewlines() -> String {
+        return components(separatedBy: .newlines).joined()
+    }
```

Comment:
If this is only used in a single place, I would just pull it into a variable where its used instead of an extension on String